roles/velero/templates/registry_gcp.yml.j2

```diff
@@ -0,0 +1,72 @@
+---
+{% for bsl in backup_storage_locations %}
+{% if bsl.provider == 'gcp' %}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  namespace: {{ velero_namespace }}
+  name: gcp-registry
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      component: gcp-registry
+  template:
+    metadata:
+      labels:
+        component: gcp-registry
+    spec:
+      restartPolicy: Always
+      containers:
+      - image: quay.io/konveyor/registry:latest
+        name: gcp-registry-container
+        ports:
+        - containerPort: 5000
+          protocol: TCP
+        env:
+          - name: REGISTRY_STORAGE
+            value: gcs
+          - name: REGISTRY_STORAGE_GCS_BUCKET
+            value: "{{ bsl.object_storage.bucket }}"
+          - name: REGISTRY_STORAGE_GCS_KEYFILE
+            value: "{{ gcp_secret_key_file }}"
+{% if bsl.config.root_directory is defined %}
+          - name: REGISTRY_STORAGE_GCS_ROOTDIRECTORY
+            value: {{ bsl.config.root_directory }}
+{% endif %}
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    component: gcp-registry
+  name: gcp-registry-svc
+  namespace: {{ velero_namespace }}
+spec:
+  ports:
+  - name: 5000-tcp
+    port: 5000
+    targetPort: 5000
+    protocol: TCP
+  selector:
+    component: gcp-registry
+  type: ClusterIP
+
+---
+apiVersion: route.openshift.io/v1
+kind: Route
```

Comment:
I'm not sure we need new route definitions for AWS+GCP+Azure. The svc + route objects are identical no matter which cloud provider is being used. We should just rename these into separate templates `registry_svc_and_route.yml.j2`. This way if we ever have to change them we only have to update one spot.

Reply:
Moved the svc + route from each provider template to an unified one. Updated the changes